mitiq/pec/decomposition/optimal.py

```diff
@@ -0,0 +1,112 @@
+    single-qubit ``ideal_operation``, the optimal decomposition is as follows:
+
+    .. math::
+         \mathcal{U}_{\beta} = \eta_1 \mathcal{O}_1 + \eta_2 \mathcal{O}_2 +
+                               \eta_3 \mathcal{O}_3 + \eta_4 \mathcal{O}_4
+
+    .. math::
+        \eta_1 =1 + \frac{3}{4} \frac{\epsilon}{1- \epsilon},
+        \qquad \mathcal{O}_1 = \mathcal{D} \circ \mathcal{I} \circ \mathcal{U}
+
+        \eta_2 =- \frac{1}{4}\frac{\epsilon}{1- \epsilon} , \qquad
+        \mathcal{O}_2 = \mathcal{D} \circ \mathcal{X} \circ \mathcal{U}
+
+        \eta_3 =- \frac{1}{4}\frac{\epsilon}{1- \epsilon} , \qquad
+        \mathcal{O}_3 = \mathcal{D} \circ \mathcal{Y} \circ \mathcal{U}
+
+        \eta_4 =- \frac{1}{4}\frac{\epsilon}{1- \epsilon} , \qquad
+        \mathcal{O}_4 = \mathcal{D} \circ \mathcal{Z} \circ \mathcal{U}
+
+    Args:
+        ideal_operation: The input ideal operation (gate + qubit) to decompose.
+        noise_level: The noise level (as a float) of the depolarizing channel.
+
+    Returns:
+        The quasi-probability representation (QPR) of the ``ideal_operation``,
+        encoded as a list of tuples, where the first element in each tuple is
+        a float coefficient, and the second element is a list of Cirq
+        ``Operation`` objects to replace the ``ideal_operation`` with.
+
+    .. note::
+        In the description we say "noisy implementable operation" but we return
+        lists of operations. This is a subtle point, relying on our assumption
+        that we can implement *any* ``ideal_operation`` followed by a single
+        depolarizing noise channel. When running on a simulator or QPU, this
+        assumption breaks down for high ``noise_level`` values.
+
+    .. [Temme2017] : Kristan Temme, Sergey Bravyi, Jay M. Gambetta,
+        "Error mitigation for short-depth quantum circuits,"
+        *Phys. Rev. Lett.* **119**, 180509 (2017),
+        (https://arxiv.org/abs/1612.02058).
+
+    .. [Takagi2020] : Ryuji Takagi,
+        "Optimal resource cost for error mitigation,"
+        (https://arxiv.org/abs/2006.12509).
+    """
+    qubits = ideal_operation.qubits
+
+    # the single-qubit case: linear combination of 1Q Paulis
+    if len(qubits) == 1:
+        q = ideal_operation.qubits[0]
+
+        epsilon = 4 / 3 * noise_level
+        alpha_pos = 1 + ((3 / 4) * epsilon / (1 - epsilon))
+        alpha_neg = -(1 / 4) * epsilon / (1 - epsilon)
+
+        alphas = [alpha_pos] + 3 * [alpha_neg]
+        post_ops = [[P(q)] for P in PAULIS]
+
+    # the two-qubit case: linear combination of 2Q Paulis
+    elif len(qubits) == 2:
+        q0, q1 = qubits
+
+        epsilon = 16 / 15 * noise_level
+        alpha_pos = 1 + ((15 / 16) * epsilon / (1 - epsilon))
+        alpha_neg = -(1 / 16) * epsilon / (1 - epsilon)
+
+        alphas = [alpha_pos] + 15 * [alpha_neg]
+        post_ops = [[Pi(q0), Pj(q1)] for Pi in PAULIS for Pj in PAULIS]
```

Comment:
Same comment as before. It is perhaps better to "do nothing" for the first coefficient  `alpha_pos` instead of applying `[I(q0), I(q1)]`. 
And probably the same should be done also for local identities: e.g.,  `[I(q0), X(q1)] `---> `[X(q1)]`.